### PR TITLE
MEN-7112: Add support for Yocto 5.0 "scarthgap"

### PR DESCRIPTION
**Pull request by TheYoctoJester**

### CODE REVIEW

**Comment by kacf** [comment]
Started an "early feedback" CI run [here](https://gitlab.com/Northern.tech/Mender/mender-qa/-/pipelines/1262862351).

**Comment by kacf** [comment]
Started another one [here](https://gitlab.com/Northern.tech/Mender/mender-qa/-/pipelines/1263923988) (first one failed because of operator error 🙂).

**Comment by kacf** [comment]
@TheYoctoJester: The error I'm now getting in the CI is of this nature:

```
ERROR: /builds/Northern.tech/Mender/meta/recipes-core/packagegroups/packagegroup-rust-cross-canadian.bb: Please ensure that your setting of VIRTUAL-RUNTIME_init_manager (systemd) matches the entries enabled in DISTRO_FEATURES
```

Does it ring a bell for you? We do have [this section](https://github.com/mendersoftware/meta-mender/blob/master/meta-mender-core/classes/mender-setup-systemd.inc#L5-L8), which looks related, but it should be active in your build as well, so I don't know why you don't see it.

Are you using `INHERIT += "mender-full"` in your build?

**Comment by TheYoctoJester** [comment]
@kacf strange. I have `systemd` enabled (https://github.com/mendersoftware/meta-mender-community/blob/scarthgap/kas/include/mender-base.yml#L31), and `INHERIT += mender-full` for all non-UBI cases (https://github.com/mendersoftware/meta-mender-community/blob/scarthgap/kas/include/mender-full.yml#L8).

Is there a way to see the effective `DISTRO_FEATURES` on that build?

**Comment by ColinFinck** [comment]
For the past few days, I have been successfully running Mender on Yocto Scarthgap with my branch at https://github.com/enlyze/meta-mender/tree/feature/scarthgap-support-for-master

Feel free to cherry-pick what is still missing.
I particularly recommend https://github.com/enlyze/meta-mender/commit/bb75c834e41152e9b0b018b80713acb62cd4f9cb or smooth upgrades from Yocto 3.1 Dunfell to Yocto 5.0 Scarthgap won't be possible.

**Comment by ColinFinck** [comment]
@TheYoctoJester:
> Is there a way to see the effective `DISTRO_FEATURES` on that build?

You are probably looking for running bitbake on your target image with the `-e` flag. That will print a full list of variables, along with their history and their final value. Among them is `DISTRO_FEATURES`

**Comment by TheYoctoJester** [comment]
> @TheYoctoJester:
> 
> > Is there a way to see the effective `DISTRO_FEATURES` on that build?
> 
> You are probably looking for running bitbake on your target image with the `-e` flag. That will print a full list of variables, along with their history and their final value. Among them is `DISTRO_FEATURES`

@ColinFinck heh yeah that one is obvious. The problem is, it works in my builds but seems to fail in the automated test runs that the team runs on GitLab. So the question is about seeing the effective `DISTRO_FEATURES` there, where the build is not interactively accessible.

**Comment by kacf** [comment]
I think I found the issue here, our build files are using the old `_append` syntax for this specific assignment. I guess it still worked on kirkstone as a backwards compatibility thing, but no more on scarthgap. I don't have time now, but I'll try another build tomorrow after fixing it.

**Comment by kacf** [comment]
> I think I found the issue here, our build files are using the old `_append` syntax for this specific assignment. I guess it still worked on kirkstone as a backwards compatibility thing, but no more on scarthgap. I don't have time now, but I'll try another build tomorrow after fixing it.

Nope, I was way off here, we are using the right syntax.

But in any case, I did figure out the reason, and I took the liberty of pushing to this branch. @TheYoctoJester, can you check the commit I added, which has some implications, and see if you agree with my reasoning?

**Comment by kacf** [comment]
New full CI pipeline [here](https://gitlab.com/Northern.tech/Mender/mender-qa/-/pipelines/1277083551).

**Comment by TheYoctoJester** [comment]
> But in any case, I did figure out the reason, and I took the liberty of pushing to this branch. @TheYoctoJester, can you check the commit I added, which has some implications, and see if you agree with my reasoning?

Hi @kacf, interesting catch. From the pure Yocto point of view, this approach is definitely to be preferred, as it requires users to make conscious decisions instead of "hidden" magic.

**Comment by kacf** [comment]
Yet [another one](https://gitlab.com/Northern.tech/Mender/mender-qa/-/pipelines/1277373315).

**Comment by lluiscampos** [comment]
New build @ [Pipeline 1280026263](https://gitlab.com/Northern.tech/Mender/mender-qa/-/pipelines/1280026263)

**Comment by lluiscampos** [comment]
New build @ [Pipeline 1281627085](https://gitlab.com/Northern.tech/Mender/mender-qa/-/pipelines/1281627085)

**Comment by svenschwermer** [comment]
How close is this PR from being merged? When can we expect scarthgap support on master?

**Comment by ruben-sch** [comment]
Any updates on this?

**Comment by TheYoctoJester** [comment]
Hi @ruben-sch @svenschwermer and all, we have given it a deeper inspection by now and unfortunately it is a bit more complex. You can find the tasks to be tackled at https://northerntech.atlassian.net/issues/?jql=labels%20%3D%20%22yocto-scarthgap%22.

We have planned the resource allocation such that everything should be finished in the first half of July 2024.

**Comment by lluiscampos** [comment]
Rebased on top of latest `master`

**Review by lluiscampos — APPROVED** [review]

File: meta-mender-demo/recipes-core/packagegroups/packagegroup-core-boot.bbappend [hunk 1]
```diff
@@ -11,3 +11,6 @@ RDEPENDS:${PN}:append:mender-update-install = " lsb-ld"
 
 # Install Mender add-ons.
 RDEPENDS:${PN}:append:mender-image = " mender-connect mender-configure mender-configure-demo mender-configure-scripts"
+
+# Except for vexpress-qemu-flash platform, where we are running out of space
+RDEPENDS:${PN}:remove:mender-image:vexpress-qemu-flash = "mender-connect mender-configure mender-configure-demo mender-configure-scripts"
```
Comment (kacf) [inline meta-mender-demo/recipes-core/packagegroups/packagegroup-core-boot.bbappend:16]: Looks good to me! Testing the client is the main importance on the Flash based setup, and expecially mender-connect is quite big.

**Review by kacf — APPROVED** [review]

**Pull request merged**